src/theory/quantifiers/ceg_t_instantiator.cpp

```diff
@@ -835,7 +842,174 @@ bool EprInstantiator::processEqualTerms( CegInstantiator * ci, SolvedForm& sf, N
   return false;
 }
 
-bool BvInstantiator::processAssertion( CegInstantiator * ci, SolvedForm& sf, Node pv, Node lit, unsigned effort ) {
+Node BvInverter::getSolveVariable( TypeNode tn ) {
+  std::map< TypeNode, Node >::iterator its = d_solve_var.find( tn );
+  if( its==d_solve_var.end() ){
+    Node k = NodeManager::currentNM()->mkSkolem( "s", tn );
+    d_solve_var[tn] = k;
+    return k;
+  }else{
+    return its->second;
+  }
+}
+
+Node BvInverter::getPathToPv( Node lit, Node pv, Node sv, std::vector< unsigned >& path, std::map< TNode, bool >& visited ) {
+  if( visited.find( lit )==visited.end() ){
+    visited[lit] = true;
+    if( lit==pv ){
+      return sv;
+    }else{
+      unsigned rmod = 0; // TODO : randomize?
+      for( unsigned i=0; i<lit.getNumChildren(); i++ ){
+        unsigned ii = ( i + rmod )%lit.getNumChildren();
+        Node litc = getPathToPv( lit[ii], pv, sv, path, visited );
```

Comment:
As you already suggested, this should probably be iterative instead of recursive.